php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -2707,10 +2707,15 @@ abstract class NDB_BVL_Instrument extends NDB_Page
      */
     public function handle(ServerRequestInterface $request) : ResponseInterface
     {
-        if ($request->getMethod() === "POST") {
+        if ($request->getMethod() === "POST" && !isset($_POST['ClearInstrument'])) {
             $this->save();
         }
-
+        if (isset($_POST['ClearInstrument'])
+            && isset($_POST['deleteconfirm'])
+            && isset($_POST['deleteconfirm2'])
+        ) {
+            $this->ClearInstrument();
```

Comment:
This function is being called from the NDB_BVL_InstrumentStatus_ControlPanel.class.inc

I dont think this is the correct fix. the problem sounds like an order of operations issue

Reply:
@ridz1208 @kongtiaowang  the `$this->ClearInstrument();` is also called in NDB_BVL_Instrument.class.inc on line 2717 if that's helpful.

Reply:
@ridz1208 @maltheism Why I changed the NDB_BVL_Instrument.class.int not NDB_BVL_InstrumentStatus_ControlPanel.class.inc ? Reason 1: They all use the same ClearInstrument function in NDB_BVL_Instrument.class.inc on line 2717. Reason 2: When the clear button clicks in ControlPanel, we need the instrument page change, not the ControlPanel. If we call this function in ControlPanel, it will not change instrument page's front-end. We have to refresh the whole page again to see the change.

Reply:
@ridz1208 what do you think about @kongtiaowang's comment? Seems to make sense to me. Although it seems weird to have the same function inside of the save method.